index.js

```diff
@@ -1,78 +1,92 @@
+        await this.loadImage(filesystemURI, this.props.source.uri);
     }
-    await this.loadImage(filesystemURI, this.props.source.uri);
-  }
 
-  async getImageFilesystemKey(remoteURI) {
-    const hashed = await Crypto.digestStringAsync(
-      Crypto.CryptoDigestAlgorithm.SHA256,
-      remoteURI
-    );
-    return `${FileSystem.cacheDirectory}${hashed}`;
-  }
+    async getImageFilesystemKey(remoteURI) {
+        const hashed = await Crypto.digestStringAsync(
+            Crypto.CryptoDigestAlgorithm.SHA256,
+            remoteURI
+        );
+        return `${FileSystem.cacheDirectory}${hashed}`;
+    }
 
-  async loadImage(filesystemURI, remoteURI) {
-    try {
-      // Use the cached image if it exists
-      const metadata = await FileSystem.getInfoAsync(filesystemURI);
-      if (metadata.exists) {
-        this.setState({
-          imgURI: filesystemURI
-        });
-        return;
-      }
+    async loadImage(filesystemURI, remoteURI) {
+        try {
+            // Use the cached image if it exists
+            const metadata = await FileSystem.getInfoAsync(filesystemURI);
+            if (metadata.exists) {
+                this.setState({
+                    imgURI: filesystemURI,
+                    loading: false
+                });
+                return;
+            }
 
-      // otherwise download to cache
-      const imageObject = await FileSystem.downloadAsync(
-        remoteURI,
-        filesystemURI
-      );
-      this.setState({
-        imgURI: imageObject.uri
-      });
+            // otherwise download to cache
+            const imageObject = await FileSystem.downloadAsync(
+                remoteURI,
+                filesystemURI
+            );
+            this.setState({
+                imgURI: imageObject.uri,
+                loading: false
+            });
+        } catch (err) {
+            console.log('Image loading error:', err);
+            this.setState({imgURI: remoteURI});
+        }
     }
-    catch (err) {
-      console.log('Image loading error:', err);
-      this.setState({ imgURI: remoteURI });
-    }
-  }
 
-  render() {
-    return (
-      <View>
-        {this.props.isBackground ? (
-          <ImageBackground
-            {...this.props}
-            source={this.state.imgURI ? { uri: this.state.imgURI } : null}
-          >
-            {this.props.children}
-          </ImageBackground>
-        ) : (
-          <Image
-            {...this.props}
-            source={this.state.imgURI ? { uri: this.state.imgURI } : null}
-          />
-        )}
-      </View>
-    );
-  }
+    render() {
+        if (this.state.loading && this.props.showActivityIndicator) {
+            // while the image is being checked and downloading
+            return (
+                <View style={this.props.activityIndicatorStyle}>
+                    <ActivityIndicator
+                        color={this.props.activityIndicatorColor ? this.props.activityIndicatorColor : '#42C2F3'}
```

Comment:
Rather than forwarding activity indicator props one by one, I think it would be good to pass a parent prop object down. The api should probably be something like `activityIndicatorProps`

Reply:
Hi Wagner, 

Thank you for the comments.

Did you meant something like this? Define an object as below

 let activityIndicatorProps = {
        activityIndicatorStyle: styles.activityIndicatorStyle,
        activityIndicatorColor: 'black',
        activityIndicatorSize: 'small',
        showActivityIndicator:true
    }

and pass this object as props as  activityIndicatorProps={activityIndicatorProps}?

Reply:
Close, I think the spread operator (https://www.techiediaries.com/react-spread-operator-props-setstate/) should be used. It would look something like this: (and not be required)
```
let activityIndicatorProps = {
color: 'black',
size: 'small',
}
```

Notice that the keys are the **actual** keys